src/modules/Popup/Popup.js

```diff
@@ -0,0 +1,196 @@
+    size: PropTypes.string,
+
+    /** Element to be rendered in-place where the popup is defined. */
+    trigger: PropTypes.node,
+
+    /** Popup width */
+    width: PropTypes.string,
+  }
+
+  static defaultProps = {
+    position: 'top left',
+  }
+
+  static _meta = _meta
+  static Content = PopupContent
+  static Header = PopupHeader
+
+  state = {}
+
+  getPosition(coords) {
+    if (!coords) return
+
+    const positions = this.props.position.split(' ')
+    const style = { position: 'absolute' }
+
+    if (includes(positions, 'right')) {
+      style.right = Math.round(document.documentElement.clientWidth - (coords.right + window.pageXOffset)) + 'px'
+      style.left = 'auto'
+    }
+
+    if (includes(positions, 'left')) {
+      style.left = Math.round(coords.left + window.pageXOffset) + 'px'
+      style.right = 'auto'
+    }
+
+    if (includes(positions, 'top')) {
+      style.bottom = Math.round(document.documentElement.clientHeight - (coords.top + window.pageYOffset)) + 'px'
+      style.top = 'auto'
+    }
+
+    if (includes(positions, 'bottom')) {
+      style.top = Math.round(coords.bottom + window.pageYOffset) + 'px'
+      style.bottom = 'auto'
+    }
+
+    return style
+  }
+
+  onOpen = (event) => {
+    this.setState({
+      coords: findDOMNode(this).getBoundingClientRect(),
+    })
+  }
+
+  componentWillMount() {
+    const portalProps = {}
+
+    if (this.props.click) {
+      portalProps.openOnTriggerClick = true
+      portalProps.closeOnDocumentClick = true
+    }
+    if (this.props.hover) {
+      portalProps.openOnTriggerMouseOver = true
+      portalProps.closeOnTriggerMouseLeave = true
+    }
+    if (this.props.focus) {
+      portalProps.openOnTriggerFocus = true
+      portalProps.closeOnTriggerBlur = true
+    }
+
+    portalProps.onOpen = this.onOpen
+
+    this.setState({ portalProps })
+  }
+
+  renderContent() {
+    const children = []
+    let index = 0
+    const { header, content } = this.props
+
+    if (header) {
+      children.push(<PopupHeader key={index++}>{header}</PopupHeader>)
+    }
+
+    if (content) {
+      if (isString(content)) {
+        children.push(<PopupContent key={index++}>{content}</PopupContent>)
+      } else {
+        children.push(content)
+      }
+    }
+
+    return children
+  }
+
+  render() {
+    const {
+      basic,
+      children,
+      className,
+      flowing,
+      fluid,
+      inverted,
+      position,
+      size,
+      trigger,
+      width,
+    } = this.props
+
+    const { coords } = this.state
+
+    const classes = cx(
+      className,
+      position,
+      size,
+      width,
+      'ui popup transition visible',
+      {
+        basic,
+        flowing,
+        fluid,
+        inverted,
+      }
+    )
+
+    return (
+      <Portal trigger={trigger} {...this.state.portalProps}>
+        <div
+          style={this.getPosition(coords)}
+          className={classes}
+        >
+        { children || this.renderContent() }
```

Comment:
Pending adding shorthand `create()` methods to the sub components, `this.renderChildren()` here can be replaced with this:

```jsx
render() {
  return (
    <Portal trigger={trigger} {...portalProps}>
      <div style={this.getStyle(coords)} className={classes}>
        {children}
        {!children && PopupHeader.create(header)}
        {!children && PopupContent.create(content)}
      </div>
    </Portal>
  )
}
```

The `create()` shorthand can be created by importing `createShorthandFactory()` and adding a static method to each component:

```jsx
import { createShorthandFactory } from '../../lib'

// PopupHeader.js
PopupHeader.create = createShorthandFactory(PopupHeader, value => ({ children: value }))

// PopupContent.js
PopupContent.create = createShorthandFactory(PopupContent, value => ({ children: value }))
```

The second argument is the `mapValueToProps()` function.  When `PopupHeader.create('my header')` is called, that function will map the string to `children`.  Factories also know how to handle numbers, props objects, and elements as a value.  So, your props will be able to handle many types of data.  You can read more in `factories.js` and see it in other components.  Again, more docs on the way for this, #561!